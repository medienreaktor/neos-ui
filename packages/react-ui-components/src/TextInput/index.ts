import {themr} from '@friendsofreactjs/react-css-themr';

import identifiers from '../identifiers';
import TextInput from './textInput';
import style from './style.css';


export default themr(identifiers.textInput, style)(TextInput);
