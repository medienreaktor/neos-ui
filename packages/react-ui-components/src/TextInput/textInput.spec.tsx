import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

import TextInput, {TextInputProps} from './textInput';

describe('<TextInput/>', () => {
    const props: TextInputProps = {
        className: 'fooClassName',
        theme: {
            'textInput': 'textInputClassName',
            'textInput--disabled': 'disabledClassName',
        },
    };

    it('should render correctly.', () => {
        const wrapper = shallow(<TextInput {...props}/>);

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should allow the propagation of "className" with the "className" prop.', () => {
        const wrapper = shallow(<TextInput {...props} className="fooClassName"/>);
        const input = wrapper.find('input');

        expect(input.prop('className')).toContain('fooClassName');
    });

    it('should call the passed "onChange" prop with the value of the input when changing it.', () => {
        const onChange = jest.fn();
        const wrapper = shallow(<TextInput {...props} onChange={onChange}/>);
        const input = wrapper.find('input');

        input.simulate('change', {
            target: {
                value: 'my value'
            }
        });

        expect(onChange.mock.calls.length).toBe(1);
        expect(onChange.mock.calls[0][0]).toBe('my value');
    });

    it('should throw no error if no "onChange" prop was passed when changing the value of the input.', () => {
        const wrapper = shallow(<TextInput {...props} />);
        const input = wrapper.find('input');

        expect(() => {
            input.simulate('change', {
                target: {
                    value: 'my value'
                }
            });
        }).not.toThrow();
    });
});
