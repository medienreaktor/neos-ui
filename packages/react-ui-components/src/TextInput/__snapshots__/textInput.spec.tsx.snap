// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput/> should render correctly. 1`] = `
<div>
  <input
    aria-disabled="false"
    aria-multiline="false"
    className="textInputClassName fooClassName"
    onChange={[Function]}
    onKeyPress={[Function]}
    role="textbox"
  />
</div>
`;
